src/base/bittorrent/session.cpp

```diff
@@ -1090,6 +1173,70 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
     settingsPack.set_bool(libt::settings_pack::enable_lsd, isLSDEnabled());
 }
 
+void Session::configurePeerClasses()
+{
+    libt::ip_filter f;
+    f.add_rule(libt::address_v4::from_string("0.0.0.0")
+               , libt::address_v4::from_string("255.255.255.255")
+               , 1 << libt::session::global_peer_class_id);
+#if TORRENT_USE_IPV6
+    f.add_rule(libt::address_v6::from_string("::0")
+               , libt::address_v6::from_string("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
+               , 1 << libt::session::global_peer_class_id);
+#endif
+    if (ignoreLimitsOnLAN()) {
```

Comment:
>I am little confused by the docs.

Approipriate docs are very incomplete and poorly (apparently, not finished yet). I had to watch code. I haven't tried to figure it out, I just looked how it works in deprecated layer.
Of course, for more detail you can see `session_impl::set_peer_classes` (assigning classes to peer) and `session_impl::incoming_connection` (using peer classes for limiting).